include/SFML/Window/Event.hpp

```diff
@@ -275,12 +267,10 @@ class SFML_WINDOW_API Event
     };
 
     ////////////////////////////////////////////////////////////
-    /// \brief Default constructor
-    ///
-    /// Sets the event to sf::Event::Empty
+    /// \brief Deleted default constructor
     ///
     ////////////////////////////////////////////////////////////
-    Event() = default;
+    Event() = delete;
```

Comment:
```suggestion
```
This can be removed since it is already implicitly deleted. To ensure we don't regress in this respect we should add a corresponding test.

https://github.com/search?q=repo%3ASFML%2FSFML%20is_default_constructible_v&type=code

I added tests for `!std::is_default_constructible_v<sf::Foo>` for all types for which we removed their default constructor so making this change is in line with existing precedent.

Reply:
Hmm, do you actually think it would be better to not have the explicit `= delete`? I think it conveys intent, and we do the same thing for some classes that cannot be copied or moved because of some of their members being non-copyable or non-movable. Technically the parent class doesn't need their copy/move constructors deleted, but we are still explicit about it.

I'd prefer leaving the explicit `= delete` in, but not a big deal for me if you believe it's best to rely on the implicit deletion.

Reply:
What I'd say is that the tests also convey intent but do so in a more terse manner. The type trait tests already cover many implicitly defined operations. Here's a good example of that: https://github.com/SFML/SFML/pull/2874

Reply:
> What I'd say is that the tests also convey intent but do so in a more terse manner.

The tests do not appear in the public API or documentation of `sf::Event`. I still think it's worth keeping the deleted ctor, but I'd rather remove it and resolve this conversation and potentially re-introduce it later to avoid friction.

Reply:
Removed the ctor and added a test